applications/nrf5340_audio/src/audio/streamctrl.c

```diff
@@ -0,0 +1,413 @@
+			ret = led_blink(LED_APP_1_BLUE);
+			ERR_CHK(ret);
+			break;
+
+		case STATE_STREAMING:
+			LOG_INF("Pausing stream");
+			audio_codec_stop();
+			ret = le_audio_pause();
+			if (ret) {
+				LOG_WRN("Failed to pause stream");
+				break;
+			}
+			stream_state_set(STATE_PAUSED);
+			ret = led_on(LED_APP_1_BLUE);
+			ERR_CHK(ret);
+			break;
+
+		default:
+			/* We got the play/pause button in a state where we
+			 * can not use it. This should be expected, buttons
+			 * can be pressed at any time. For now, inform about it.
+			 */
+			LOG_DBG("Got play/pause in state %d", m_stream_state);
+		}
+		break;
+
+	case BUTTON_VOLUME_UP:
+		le_audio_volume_up();
+		break;
+
+	case BUTTON_VOLUME_DOWN:
+		le_audio_volume_down();
+		break;
+
+	case BUTTON_MUTE:
+		le_audio_volume_mute();
+		break;
+
+	case BUTTON_TEST_TONE:
+		switch (m_stream_state) {
+		case STATE_STREAMING:
+			if (CONFIG_AUDIO_BIT_DEPTH_BITS != 16) {
+				LOG_WRN("Tone gen only supports 16 bits");
+				break;
+			}
+
+			if (test_tone_hz == 0) {
+				test_tone_hz = TEST_TONE_BASE_FREQ_HZ;
+			} else if (test_tone_hz >= TEST_TONE_BASE_FREQ_HZ * 4) {
+				test_tone_hz = 0;
+			} else {
+				test_tone_hz = test_tone_hz * 2;
+			}
+
+			LOG_INF("Test tone set at %d Hz", test_tone_hz);
+
+			ret = audio_encode_test_tone_set(test_tone_hz);
+			ERR_CHK_MSG(ret, "Failed to generate test tone");
+			break;
+
+		default:
+			LOG_WRN("Test tone can only be set in streaming mode");
+			break;
+		}
+		break;
+
+	default:
+		LOG_WRN("Unexpected/unhandled button id: %d", event.button_pin);
+	}
+}
+
+/* Handle Bluetooth LE audio events */
+static void m_le_audio_evt_handler(enum le_audio_evt_type event)
+{
+	int ret;
+
+	LOG_DBG("Received event = %d, current state = %d", event, m_stream_state);
+	switch (event) {
+	case LE_AUDIO_EVT_STREAMING:
+		LOG_INF("LE audio evt streaming");
+		audio_codec_start();
+		stream_state_set(STATE_STREAMING);
+		ret = led_blink(LED_APP_1_BLUE);
+		ERR_CHK(ret);
+		break;
+
+	case LE_AUDIO_EVT_NOT_STREAMING:
+		LOG_INF("LE audio evt not_streaming");
+		audio_codec_stop();
+		stream_state_set(STATE_PAUSED);
+		ret = led_on(LED_APP_1_BLUE);
+		ERR_CHK(ret);
+		break;
+
+	default:
+		LOG_WRN("Unexpected/unhandled event: %d", event);
+		break;
+	}
+}
+
+void streamctrl_event_handler(void)
+{
+	struct event_t my_event;
+
+	/* As long as this timeout is K_FOREVER, ctrl_events_get should
+	 * never return unless it has an event, that is why we can ignore the
+	 * return value
+	 */
+	(void)ctrl_events_get(&my_event, K_FOREVER);
+
+	switch (my_event.event_source) {
+	case EVT_SRC_BUTTON:
+		m_button_evt_handler(my_event.button_activity);
+		break;
+
+	case EVT_SRC_LE_AUDIO:
+		m_le_audio_evt_handler(my_event.le_audio_activity.le_audio_evt_type);
+		break;
+
+	default:
+		LOG_WRN("Unhandled event from queue -  source = %d", my_event.event_source);
+	}
+}
+
+/**** The top level ****/
+/* Initialize what must be initialized, to be ready to handle events */
```

Comment:
Change to a more describing function name? Like streamctrl_init_start()

Reply:
If audio_system_init() is moved to main, the .h file documentation could be something like:
"Init internal functionality and start streamctrl"